### PR TITLE
Update versioning-policy.md to add the type addition condition

### PR DESCRIPTION
**Pull request by curquiza**
Following this internal discussion: https://meilisearch.slack.com/archives/CD7Q2UKGB/p1670853892183669

@brunoocasali and @irevoire I would like to have your review on this 😇

### CODE REVIEW

File: resources/versioning-policy.md [hunk 1]
```diff
@@ -52,6 +52,12 @@ This kind of change will lead us to increase the MINOR to let the people know ab
 
 🚨 Any change about the relevancy that is related to API usage, and thus, that may impact users to change their code (for instance changing the default `matchingStrategy` value) is not related to this specific section and would lead us to increase the MAJOR.
 
+### New "variant" type addition
+
+We don't consider breaking to add a new type to an already existing list of variant. For example, adding a new type of `task`, or a new type of error `code`.
+
+We are aware some strongly type language code base could be impacted, and we always would recommend setting an `unknown` type when working with strongly typed language and Meilisearch.
```
Comment (brunoocasali) [inline resources/versioning-policy.md:59]: A suggestion about the strongly typed stacks...
```suggestion
We are aware some strongly typed language code bases could be impacted, and our recommendation is to set an `unknown` type when possible.
```

**Review by brunoocasali — APPROVED** [review]

File: resources/versioning-policy.md [hunk 2]
```diff
@@ -52,6 +52,12 @@ This kind of change will lead us to increase the MINOR to let the people know ab
 
 🚨 Any change about the relevancy that is related to API usage, and thus, that may impact users to change their code (for instance changing the default `matchingStrategy` value) is not related to this specific section and would lead us to increase the MAJOR.
 
+### New "variant" type addition
+
+We don't consider breaking to add a new type to an already existing list of variant. For example, adding a new type of `task`, or a new type of error `code`.
+
+We are aware some strongly typed language code bases could be impacted, and our recommendation is to set an `unknown` type when possible.
```
Comment (irevoire) [inline resources/versioning-policy.md:59]:
```suggestion
We are aware some strongly typed language code bases could be impacted, and our recommendation is to handle the possibility of an unknown type when deserializing the answers of meilisearch.
```

I feel like your sentence is giving a solution on how to handle this instead of just letting people know it will happen :thinking: 
I don't like the formulation of mine either, but I feel like it's easier to understand where the issue will arise, and then it lets you conceive your own solution how you feel it should work with your language (for example, in rust it won't be a type but simply an enum variant, while in C# it's probably going to be a class?)

tldr: I don't like the word `set` and the fact that `unknown` is in a code block :joy:

Comment (curquiza) [reply resources/versioning-policy.md:59]:
```suggestion
We are aware some strongly typed language code bases could be impacted, and our recommendation is to handle the possibility of an unknown type when deserializing Meilisearch's response.
```

Comment (curquiza) [reply resources/versioning-policy.md:59]: I like your suggestion, I just made some small changes 👌

**Review by irevoire — APPROVED** [review]
That's good to me :+1:

File: resources/versioning-policy.md [hunk 3]
```diff
@@ -52,6 +52,12 @@ This kind of change will lead us to increase the MINOR to let the people know ab
 
 🚨 Any change about the relevancy that is related to API usage, and thus, that may impact users to change their code (for instance changing the default `matchingStrategy` value) is not related to this specific section and would lead us to increase the MAJOR.
 
+### New "variant" type addition
+
+We don't consider breaking to add a new type to an already existing list of variant. For example, adding a new type of `task`, or a new type of error `code`.
+
+We are aware some strongly typed language code bases could be impacted, and our recommendation is to handle the possibility of an unknown type when deserializing Meilisearch's response.
```
Comment (brunoocasali) [inline resources/versioning-policy.md:59]: What about this?
```suggestion
We are aware some strongly typed language code bases could be impacted, and our recommendation is to handle the possibility of having an unknown type when deserializing Meilisearch's response.
```

I got @irevoire's idea, and I agree with it. I think adding this verb will tell the user that they can expect an unknown type and they should handle it by themselves.

**Review by brunoocasali — APPROVED** [review]

**Comment by curquiza** [comment]
For your information @meilisearch/engine-team
